octoprint_mrbeam/analytics/usage_handler.py

```diff
@@ -246,27 +301,157 @@ def _calculate_ntp_fix_compensation(self, job_duration):
 
         return job_duration
 
-    def reset_prefilter_usage(self):
-        self._usage_data["prefilter"]["job_time"] = 0
+    def _get_airfilter_usage_data(self, serial):
+        """
+        Returns the usage data for the airfilter with the given serial.
+
+        Args:
+            serial: Serial of the airfilter
+
+        Returns:
+            (dict): Usage data for the airfilter with the given serial
+        """
+        airfilter_usage_data = self._usage_data.get(self.AIRFILTER_KEY, {}).get(
+            serial, {}
+        )
+        return airfilter_usage_data if airfilter_usage_data is not None else {}
+
+    def _get_airfilter_prefilter_usage_data(self, serial=None):
+        """
+        Get the usage data for the prefilter of the airfilter with the given serial.
+
+        Args:
+            serial: Serial of the airfilter
+
+        Returns:
+            (dict): Usage data for the prefilter of the airfilter with the given serial
+        """
+        if serial is None:
+            serial = self._get_airfilter_serial()
+        return self._get_airfilter_usage_data(serial).get(self.PREFILTER_KEY, {})
+
+    def _get_airfilter_carbon_filter_usage_data(self, serial=None):
+        """
+        Get the usage data for the carbon filter of the airfilter with the given serial.
+
+        Args:
+            serial: Serial of the airfilter
+
+        Returns:
+            (dict): Usage data for the carbon filter of the airfilter with the given serial
+        """
+        if serial is None:
+            serial = self._get_airfilter_serial()
+        return self._get_airfilter_usage_data(serial).get(self.CARBON_FILTER_KEY, {})
+
+    def _set_job_time(self, component, job_time):
+        """
+        Set the job time for the given component.
+
+        Args:
+            component: Component to set the job time for
+            job_time: job time in seconds
+
+        Returns:
+            None
+        """
+        element = self._usage_data
+        if not isinstance(component, list):
+            component = [component]
+        for key in component:
+            if key not in element:
+                element[key] = {}
+                self._logger.info("Created new usage data key: {}".format(key))
+            element = element[key]
+        element[self.JOB_TIME_KEY] = job_time
+        self._logger.debug(
+            "Set job time for component {} to {} - {}".format(
+                component, job_time, self._usage_data
+            )
+        )
+
+    def _get_job_time(self, usage_data):
+        """
+        Get the job time from the given usage data.
+
+        Args:
+            usage_data: Usage data to get the job time from
+
+        Returns:
+            int: job time in seconds
+        """
+        return usage_data.get(self.JOB_TIME_KEY, 0)
```

Comment:
Isn't None better here to detect if there was no value?
```suggestion
        return usage_data.get(self.JOB_TIME_KEY, None)
```